#Introduction

Streamline Pro is a comprehensive project management and collaboration tool designed to help businesses manage projects, track progress, and collaborate efficiently. Understanding the unit economics of Streamline Pro is crucial for evaluating its financial health and sustainability. This involves analysing key metrics such as Customer Acquisition Cost (CAC), Average Revenue Per User (ARPU), Cost of Goods Sold (COGS), Gross Margin, Customer Lifetime Value (LTV), and the LTV/CAC ratio.

#Unit Economics

##Import data

In [1]:
import pandas as pd


In [2]:
def read_file(url, month_col=None):
    df = pd.read_excel('https://docs.google.com/spreadsheets/d/' + url + '/export?format=xlsx')
    if month_col is not None:
        df_03_2023 = df[(df[month_col].dt.month == 3) & (df[month_col].dt.year == 2023)]
        return df_03_2023
    else:
        return df

# Reading the data using the function
df_customer_lifespan_data = read_file(url = '1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI')
df_daily_marketing_spendings = read_file(url = '1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c', month_col = "date")
df_monthly_expenses = read_file(url = '10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw', month_col = "month")
df_payroll = read_file(url = '1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4', month_col = "month")
df_receipts_history = read_file(url = '1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE', month_col = "date")

## CAC

In [4]:
# Calculate online ads cost
online_ads_cost = df_daily_marketing_spendings['spending'].sum()

# Calculate salary for Sales and Marketing departments
sale_salary = df_payroll[df_payroll['department'].isin(["Sales", "Marketing"])]['paid'].sum()

# Calculate the marketing software cost for "Salesforce"
marketing_software_cost = df_monthly_expenses[df_monthly_expenses['item'] == "Salesforce"]['amount'].sum()

# Calculate total Sales and Marketing expenses
total_sale_mkt_expenses = online_ads_cost + sale_salary + marketing_software_cost

In [5]:
new_cus_count = df_receipts_history['new_customer'].sum()

In [6]:
CAC = total_sale_mkt_expenses/new_cus_count

In [7]:
print("CAC =",round(CAC,3))

CAC = 1213.968


##ARPU

In [8]:
revenue = df_receipts_history['receipt_amount'].sum()

In [9]:
num_cus = len(df_receipts_history['customer_id'].unique())

In [10]:
arpu = revenue/num_cus

In [ ]:
print("ARPU =",round(arpu,3))

ARPU = 284.36


##COGS

In [11]:
server_and_software_licences = df_monthly_expenses[df_monthly_expenses['item'].isin(['AWS Hosting', 'Google Cloud Storage', 'Atlassian Jira','Slack', 'Zoom'])]['amount'].sum()

In [12]:
salary_engineering = df_payroll[df_payroll['department'] == 'Engineering']['paid'].sum()

In [13]:
cogs = server_and_software_licences + salary_engineering

In [15]:
print("COGS =",cogs)

COGS = 20840


##GROSS MARGIN

In [16]:
gross_margin = (revenue - cogs)/revenue * 100

In [17]:
print("Gross Margin =",round(gross_margin,3),"%")

Gross Margin = 74.902 %


##LTV

In [30]:
# Ensure 'signup_date' and 'churn_date' are in datetime format
df_customer_lifespan_data['start_date'] = pd.to_datetime(df_customer_lifespan_data['start_date'])
df_customer_lifespan_data['churn_date'] = pd.to_datetime(df_customer_lifespan_data['churn_date'])

# Calculate the lifespan in months
df_customer_lifespan_data['lifespan_months'] = (df_customer_lifespan_data['churn_date'] - df_customer_lifespan_data['start_date']).dt.days / 30

# Calculate the average lifespan in months
avg_lifespan_months = df_customer_lifespan_data['lifespan_months'].mean()

In [32]:
ltv = arpu*gross_margin*avg_lifespan_months

In [34]:
print("LTV =",round(ltv,3),"$")

LTV = 209610.289 $


##LTV/CAC

In [36]:
ltv_per_cac = ltv/CAC

In [37]:
print("LTV/CAC =",round(ltv_per_cac,3))

LTV/CAC = 172.665


#Conclusion:

Streamline Pro has demonstrated a strong potential for sustainable growth and profitability. \
With a Customer Acquition Cost \(CAC) of \$1,213.97 and a Lifetime Value of \$209,610.29, the company is generating significant returns per customer.\
The LTV/CAC ratio of 172.67 indicates that for every dollar spent on acquiring a customer, the company can expect to generate over 172 times that amount in revenue, which is a strong sign of business efficiency and profitability.\
Moreover, with an ARPU of $284.36 and a gross margin of 774.9%, Streamline Pro is effectively managing its costs and generating a high level of value per user, while maintaining healthy profit margins.These figures highlight the company's ability to deliver both operational and financial efficiency.\
In sumary, Streamline Pro's Robust financial metrics suggest that it is on a strong growth trajectory with a profitable customer base and a solid business model. The company's effective cost management, coupled with its strong customer value proposition, positions it well for continued success in the market.